src/huggingface_hub/inference/_client.py

```diff
@@ -2392,7 +2392,13 @@ def visual_question_answering(
         return VisualQuestionAnsweringOutputElement.parse_obj_as_list(response)
 
     def zero_shot_classification(
-        self, text: str, labels: List[str], *, multi_label: bool = False, model: Optional[str] = None
+        self,
+        text: str,
+        labels: List[str],
+        *,
+        multi_label: bool = False,
+        model: Optional[str] = None,
+        hypothesis_template: str = "This example is {}.",
```

Comment:
I would not add a default value here. We don't have default hypothesis templates in either the [API specs](https://github.com/huggingface/huggingface.js/blob/main/packages/tasks/src/tasks/zero-shot-classification/spec/input.json) or the [tasks page](https://huggingface.co/tasks/zero-shot-classification). 

Could you make the arg optional + default to None + send in payload only if not None:

```suggestion
        hypothesis_template: Optional[str] = None,
```

Reply:
The NLI-based zeroshot classifiers only work properly when the labels are wrapped into a hypothesis, because the underlying NLI models were trained on full sentences. The zeroshotpipeline has the same default template "This example is {}.", see [here](https://huggingface.co/docs/transformers/en/main_classes/pipelines#transformers.ZeroShotClassificationPipeline.__call_). Is the hypothesis template applied somewhere else downstream if we don't apply it here?

Reply:
It will be applied server-side since InferenceAPI is just a wrapper to forward parameters to `transformers` (cf our offline conv)